From 6c9fa0066ca72288c31bea39c123bc0b3efbb46a Mon Sep 17 00:00:00 2001
From: lirui <lirui@unknown>
Date: Mon, 9 Mar 2015 13:14:03 +0000
Subject: [PATCH 0076/1164] HIVE-9855: Runtime skew join doesn't work when
 skewed data only exists in big table (Rui
 reviewed by Xuefu) merged from trunk

git-svn-id: https://svn.apache.org/repos/asf/hive/branches/spark@1665227 13f79535-47bb-0310-9956-ffa450edef68
---
 .../hadoop/hive/ql/exec/SkewJoinHandler.java       |   13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/exec/SkewJoinHandler.java b/ql/src/java/org/apache/hadoop/hive/ql/exec/SkewJoinHandler.java
index a63466a..74e6d15 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/exec/SkewJoinHandler.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/exec/SkewJoinHandler.java
@@ -317,14 +317,11 @@ private void commitOutputPathToFinalPath(Path specPath,
     Path outPath = getOperatorOutputPath(specPath);
     Path finalPath = getOperatorFinalPath(specPath);
     FileSystem fs = outPath.getFileSystem(hconf);
-    try {
-      if (!fs.rename(outPath, finalPath)) {
-        throw new IOException("Unable to rename output to: " + finalPath);
-      }
-    } catch (FileNotFoundException e) {
-      if (!ignoreNonExisting) {
-        throw e;
-      }
+    if (ignoreNonExisting && !fs.exists(outPath)) {
+      return;
+    }
+    if (!fs.rename(outPath, finalPath)) {
+      throw new IOException("Unable to rename output to: " + finalPath);
     }
   }
 
-- 
1.7.9.5

